prov/shm/src/smr_ep.c

```diff
@@ -319,6 +319,88 @@ void smr_format_iov(struct smr_cmd *cmd, fi_addr_t peer_id,
 	smr_post_pend_resp(pend, cmd, context, iov, count, resp);
 }
 
+void smr_format_mmap(struct smr_cmd *cmd, fi_addr_t peer_id,
+		     const struct iovec *iov, size_t count, size_t total_len,
+		     uint32_t op, uint64_t tag, uint64_t data, uint64_t op_flags,
+		     void *context, struct smr_region *smr, uint64_t msg_id,
+		     struct smr_ep_name *map_name,
+		     struct smr_resp *resp, struct smr_tx_entry *pend)
+{
+	smr_generic_format(cmd, peer_id, op, tag, 0, 0, data, op_flags);
+	cmd->msg.hdr.op_src = smr_src_mmap;
+	cmd->msg.hdr.msg_id = msg_id;
+	cmd->msg.hdr.src_data = (uintptr_t) ((char **) resp - (char **) smr);
+	cmd->msg.hdr.size = total_len;
+
+	smr_post_pend_resp(pend, cmd, context, iov, count, resp);
+
+	pend->map_name = map_name;
```

Comment:
No particular reason. The map_name was being saved at the end of post_pend_resp function before. Since we combined mmap_copy_in() and format_mmap(), I just put it after post_pend_resp call. 
I have moved it before pos_pend_resp call.